### PR TITLE
`Popup` attachment element

### PR DESCRIPTION
**Pull request by sorenoid**
support to display AttachmentsPopupElement in a Popup with
* gallery view of read only Popup attachments.
* support for loading attachments
* support for image, video, document and "other" content types.
* support for thumbnails on image attachments.
* encapsulated theming values

### CODE REVIEW

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/attachment/AttachmentsElementDefaults.kt [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.arcgismaps.toolkit.popup.internal.element.attachment
+
+
```
Comment (puneet-pdx) [inline toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/attachment/AttachmentsElementDefaults.kt:19]: Unnecessary newlines here

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/attachment/AttachmentsElementState.kt [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+            restore = {
+                AttachmentsElementState(
+                    element
+                )
+            }
+        )
+
+    }
+}
+
```
Comment (puneet-pdx) [inline toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/attachment/AttachmentsElementState.kt:77]: unnecessary newline

File: toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/attachment/AttachmentsPopupElement.kt [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    LazyRow(
+        state = state,
+        horizontalArrangement = Arrangement.spacedBy(15.dp),
+    ) {
+        items(attachments, key = { it.name + it.type + it.size }) {
+            AttachmentTile(it)
+        }
+    }
+}
+
```
Comment (puneet-pdx) [inline toolkit/popup/src/main/java/com/arcgismaps/toolkit/popup/internal/element/attachment/AttachmentsPopupElement.kt:79]: could remove the newline here

**Review by puneet-pdx — APPROVED** [review]
@sorenoid LGTM 👍🏽 . Could use some cleaning up of unnecessary newlines which I pointed out.

**Review by eri9000 — CHANGES_REQUESTED** [review]
Good work 👍🏼 
A few suggestions...

**Review by eri9000 — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**